report_builder/views.py

```diff
@@ -132,8 +131,64 @@ def async_report_save(self, report, objects_list, title, header, widths, user):
     def get(self, request, *args, **kwargs):
         report_id = kwargs['pk']
         if getattr(settings, 'REPORT_BUILDER_ASYNC_REPORT', False):
-            from .tasks import report_builder_async_report_save
-            report_task = report_builder_async_report_save.delay(
+            from .tasks import report_builder_xlsx_async_report_save
+            report_task = report_builder_xlsx_async_report_save.delay(
+                report_id, request.user.pk)
+            task_id = report_task.task_id
+            return HttpResponse(
+                json.dumps({'task_id': task_id}),
+                content_type="application/json")
+        else:
+            return self.process_report(
+                report_id, request.user.pk, to_response=True)
+
+
+class DownloadCsvView(DataExportMixin, View):
```

Comment:
This is a lot of code just for using csv instead of xlsx when all the code for csv is in report utils anyway. Could we have a generic download view with a switch to do csv instead of xlsx?